src/zenml/integrations/dash/visualizers/pipeline_run_lineage_visualizer.py

```diff
@@ -0,0 +1,344 @@
+                                                    className="me-1",
+                                                ),
+                                                dbc.Badge(
+                                                    "Running",
+                                                    color=COLOR_YELLOW,
+                                                    className="me-1",
+                                                ),
+                                                dbc.Badge(
+                                                    "Failed",
+                                                    color=COLOR_RED,
+                                                    className="me-1",
+                                                ),
+                                            ]
+                                        ),
+                                    ]
+                                ),
+                                dbc.Row(
+                                    [
+                                        cyto.Cytoscape(
+                                            id="cytoscape",
+                                            layout={
+                                                "name": "breadthfirst",
+                                                "roots": f'[id = "{first_step_id}"]',
+                                            },
+                                            elements=edges + nodes,
+                                            stylesheet=STYLESHEET,
+                                            style={
+                                                "width": "100%",
+                                                "height": "800px",
+                                            },
+                                            zoom=1,
+                                        )
+                                    ]
+                                ),
+                                dbc.Row(
+                                    [
+                                        dbc.Button(
+                                            "Reset",
+                                            id="bt-reset",
+                                            color="primary",
+                                            className="me-1",
+                                        )
+                                    ]
+                                ),
+                            ]
+                        ),
+                        dbc.Col(
+                            [
+                                dcc.Markdown(id="markdown-selected-node-data"),
+                            ]
+                        ),
+                    ]
+                ),
+            ],
+            className="p-5",
+        )
+
+        @app.callback(
+            Output("markdown-selected-node-data", "children"),
+            Input("cytoscape", "selectedNodeData"),
+        )
+        def display_data(data_list) -> str:
```

Comment:
Missing a type annotation for data list

Reply:
This one might be accurate:

```suggestion
        def display_data(data_list: List[Dict[str, Union[str, int, bool]]]) -> str:
```